src/runtime_src/tools/xclbinutil/XclBinClass.cxx

```diff
@@ -1585,62 +1605,62 @@ XclBin::setKeyValue(const std::string & _keyValue)
   throw std::runtime_error(errMsg);
 }
 
-void 
-XclBin::removeKey(const std::string & _sKey)
+void
+XclBin::removeKey(const std::string& _sKey)
 {
 
   XUtil::TRACE(XUtil::format("Removing User Key: '%s'", _sKey.c_str()));
 
-  Section *pSection = findSection(KEYVALUE_METADATA);
+  Section* pSection = findSection(KEYVALUE_METADATA);
   if (pSection == nullptr) {
     std::string errMsg = XUtil::format("ERROR: Key '%s' not found.", _sKey.c_str());
     throw std::runtime_error(errMsg);
   }
 
   boost::property_tree::ptree ptKeyValueMetadata;
-   pSection->getPayload(ptKeyValueMetadata);
-
-   XUtil::TRACE_PrintTree("KEYVALUE:", ptKeyValueMetadata);
-   boost::property_tree::ptree ptKeyValues = ptKeyValueMetadata.get_child("keyvalue_metadata");
-   std::vector<boost::property_tree::ptree> keyValues = XUtil::as_vector<boost::property_tree::ptree>(ptKeyValues, "key_values");
-
-   // Update existing key
-   bool bKeyFound = false;
-   for (unsigned int index = 0; index < keyValues.size(); ++index) {
-      if (keyValues[index].get<std::string>("key") == _sKey) {
-         bKeyFound = true;
-         XUtil::QUIET(std::string("Removing key '") + _sKey + "'");
-         keyValues.erase(keyValues.begin() + index);
-         break;
-      }
+  pSection->getPayload(ptKeyValueMetadata);
+
+  XUtil::TRACE_PrintTree("KEYVALUE:", ptKeyValueMetadata);
+  boost::property_tree::ptree ptKeyValues = ptKeyValueMetadata.get_child("keyvalue_metadata");
+  std::vector<boost::property_tree::ptree> keyValues = XUtil::as_vector<boost::property_tree::ptree>(ptKeyValues, "key_values");
```

Comment:
why repeat the type?  auto?

Reply:
I understand fully what auto does and I do use it frequency when I am working with iterators.   

That said, I do see the redundancy here and have updated the return type to be auto (as well as other places in the code base).